src/hotspot/share/cds/heapShared.cpp

```diff
@@ -179,8 +179,9 @@ void HeapShared::fixup_regions() {
 }
 
 unsigned HeapShared::oop_hash(oop const& p) {
-  unsigned hash = (unsigned)p->identity_hash();
-  return hash;
+  // Do not call p->identity_hash() as that will update the
+  // object header.
+  return primitive_hash(cast_from_oop<intptr_t>(p));
```

Comment:
This fix doesn't affect the contents of the archive, but it's necessary to avoid incorrect output from `-Xlog:cds+map=trace`.